staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -821,6 +846,9 @@ func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 			watcher.nonblockingAdd(event)
 		}
 	} else {
+		// Cache serializations of the object across all watchers.
+		c.setCachingObjects(event)
```

Comment:
is there a reason not to make the cacheableObject at the point in time we create the watchCacheEvent or storeElement? do we own the object there and could we avoid making a copy in newCachingObject in that case?

Reply:
that would also let us handle/propagate errors better instead of just logging them